utils/fhir-server-export.py

```diff
@@ -0,0 +1,151 @@
+    """Download given large file via streaming"""
+    # https://stackoverflow.com/a/16696317
+    headers = {}
+    if auth_token is not None:
+        headers["Authorization"] = f"Bearer {auth_token}"
+
+    if not filename:
+        filename = url.split("/")[-1]
+    with requests.get(url, headers=headers, stream=True) as r:
+        r.raise_for_status()
+        with open(filename, "wb") as f:
+            for chunk in r.iter_content(chunk_size=8192):
+                # If you have chunk encoded response uncomment if
+                # and set chunk_size parameter to None.
+                # if chunk:
+                f.write(chunk)
+    return filename
+
+
+def fixup_url(url, base_url):
+    """
+    Replace FHIR base URL in given FHIR API call with different base_url
+    Helpful when a FHIR server is configured with a server_name that does not match its public one
+    No generalized solution when urls significantly different
+    """
+
+    # Hapi ignores server_name configuration
+    if base_url.startswith("https://"):
+        url = url.replace("http://", "https://")
+
+    if url.startswith(base_url):
+        return url
+
+    second_last_path, last_path = url.split("/")[-2:]
+    # FHIR operations
+    if last_path.startswith("$"):
+        return f"{base_url}/{last_path}"
+
+    # known resources (Hapi FHIR uses Binary)
+    if second_last_path in ("Binary"):
+        return f"{base_url}/{second_last_path}/{last_path}"
+
+
+def poll_status(status_poll_url, auth_token=None, max_rety_time=600):
+    """Poll given status URL until ready (or timeout). Returns response JSON when ready to download"""
+
+    headers = {}
+    if auth_token is not None:
+        headers["Authorization"] = f"Bearer {auth_token}"
+
+    rety_time = 0
+    while rety_time < max_rety_time:
+        status_poll_response = requests.get(status_poll_url, headers=headers)
+        status_poll_response.raise_for_status()
+
+        retry_after = int(status_poll_response.headers.get("Retry-After", 0))
+        if not retry_after:
+            return status_poll_response.json()
+
+        progress = status_poll_response.headers.get("X-Progress")
+        if progress:
+            print("progress: ", progress)
+
+        print(f"waiting {retry_after} seconds")
+        rety_time += retry_after
+        time.sleep(retry_after)
+    print("timeout exceeded")
+    exit(1)
+
+
+def kickoff(base_url, no_cache=False, auth_token=None, type=None):
+    """Initate a Bulk Export, return endpoint to poll"""
+    headers = {
+        "Accept": "application/fhir+json",
+        "Prefer": "respond-async",
+    }
+
+    params = {}
+    if no_cache:
+        print("server-side caching disabled")
+        headers["Cache-control"] = "no-cache"
+
+    if auth_token is not None:
+        headers["Authorization"] = f"Bearer {auth_token}"
+
+    if type is not None:
+        params["_type"] = type
+
+    kickoff_response = requests.post(
```

Comment:
@mcjustin finally figured out this was why Hapi was ignoring our [parameters](https://build.fhir.org/ig/HL7/bulk-data/export.html#query-parameters): I misread the Bulk Export spec. 

support for kickoff requests via `GET` is required. `POST` kickoff requests are also allowed, but use a different format (POSTing FHIR Parameters JSON data)

>A server SHALL support GET requests and MAY support POST requests that supply parameters using the FHIR [Parameters Resource](https://www.hl7.org/fhir/parameters.html)

https://build.fhir.org/ig/HL7/bulk-data/export.html#bulk-data-kick-off-request

Doing some more testing but should have a fix soon